makefile.shade

```diff
@@ -203,15 +204,85 @@ var buildTarget = "compile"
             CloneOrUpdate(repo);
 
             GitCommand(repo, "checkout origin/release -B master");
-            File.Copy(Path.Combine("build-template", "NuGet.master.config"),
+
+            if(File.Exists(Path.Combine(repo, "NuGet.config")))
+            {
+                File.Copy(Path.Combine("build-template", "NuGet.master.config"),
                       Path.Combine(repo, "NuGet.config"),
                       overwrite: true);
-            GitCommand(repo, "commit -am \"Updating NuGet.config\"");
+                GitCommand(repo, "commit -am \"Updating NuGet.config\"");
+            }
 
             GitCommand(repo, "push origin master:master -f");
         }
     }
 
+#pin-version
+    @{
+        // NOTE: Before running update the below monikers as needed
+        // TODO: Should these be taken from environmental variable ?
+        var branch = "master";
+        var oldPreRelease = "";
+        var newPreRelease = "";
+        var excludeReposForJson = new[]
+            {
+              "Coherence",
+              "Coherence-Signed",
+              "dnvm",
+              "Entropy"
+            };
+
+        foreach (var repo in GetAllRepos().Except(excludeReposForJson))
+        {
+            CloneOrUpdate(repo);
+
+            GitCommand(repo, string.Format("checkout {0}", branch));
+
+            // Process project.json files to pin version numbers
+            // Replace version in project.json
+            foreach(var file in Directory.GetFiles(repo,"project.json",SearchOption.AllDirectories))
+            {
+                var builder = new StringBuilder();
+                var contents = File.ReadAllLines(file);
+
+                foreach(var jsonEntry in contents)
+                {
+                    if(!jsonEntry.Contains("\"System."))
+                    {
+                        var oldSearchExp = string.IsNullOrEmpty(oldPreRelease) ? "0.0-*" : string.Format("0.0-{0}-*" , oldPreRelease);
+                        var newReplaceExp = string.IsNullOrEmpty(newPreRelease) ? "0.0-*" : string.Format("0.0-{0}-*" , newPreRelease);
+
+                        var newEntry = ProcessExcludesForPackages(jsonEntry);
+
+                        builder.AppendLine(newEntry.Replace(oldSearchExp, newReplaceExp));
+                    }
+                    else
+                    {
+                        builder.AppendLine(jsonEntry);
+                    }
+                }
+
+                File.WriteAllText(file,builder.ToString());
+            }
+
+            // Build projects to produce project.lock.json
+            Exec("build.cmd", "compile", repo);
+
+           // Replace lock=true in project.lock.json
+            foreach(var file in Directory.GetFiles(repo,"project.lock.json",SearchOption.AllDirectories))
+            {
+                var contents = File.ReadAllText(file);
```

Comment:
`dnu` is not found in the path so I need to send it with the path in the `Exec` function call. Now I couldn't figure out a easy way of getting the path apart from reading a env variable. Any pointers would help for sure